README.md

```diff
@@ -12,62 +51,164 @@ The recommended way to install is through Composer:
 composer require fakturoid/fakturoid-php
 ```
 
-Library requires PHP 5.3.0 (or later) and `ext-curl` and `ext-json` extensions.
+Library requires PHP 8.1 (or later) and `ext-json`, `nyholm/psr7` and `psr/http-client` extensions.
+
+## Authorization by OAuth 2.0
+
+### Authorization Code Flow
+
+Authorization using OAuth takes place in several steps. We use data obtained from the developer portal as client ID and client secret (_Settings → Connect other apps → OAuth 2 for app developers_).
+
+First, we offer the user a URL address where he enters his login information. We obtain this using the following method:
+```php
+$fManager = new \Fakturoid\FakturoidManager(
+    new ClientInterface(), // PSR-18 client
+    '{fakturoid-client-id}',
+    '{fakturoid-client-secret}',
+    'PHPlib <your@email.cz>',
+    null,
+    '{your-redirect-uri}'
+);
+echo '<a href="' . $fManager->getAuthenticationUrl() . '">Link</a>';
+```
+After entering the login data, the user is redirected to the specified redirect URI and with the code with which we obtain his credentials. We process the code as follows:
+```php
+$fManager->requestCredentials($_GET['code']);
+```
+Credentials are now established in the object instance and we can send queries to the Fakturoid api. Credentials can be obtained in 2 ways. Obtaining credentials directly from the object:
+```php
+$credentials = $fManager->getCredentials();
+echo $credentials->toJson();
+```
+
+### Client Credentials Flow
+
+```php
+$fManager = new \Fakturoid\FakturoidManager(
+    new ClientInterface(), // PSR-18 client
+    '{fakturoid-client-id}',
+    '{fakturoid-client-secret}',
+    'PHPlib <your@email.cz>'
+);
+$fManager->authClientCredentials();
+```
+
+## Processing credentials using the credentials callback:
+The way callback works is that the library calls the callback function whenever the credentials are changed. This is useful because the token is automatically refreshed after its expiration.
+```php
+$fManager->setCredentialsCallback(new class implements \Fakturoid\Auth\CredentialCallback {
+    public function __invoke(?\Fakturoid\Auth\Credentials $credentials = null): void
+    {
+        // Save credentials to database or another storage
+    }
+});
+```
+
 
 ## Usage
 
+### Upload credentials to the Fakturoid manager
```

Comment:
Nějak se mi nelíbí to slovo upload. Co kdybychom zkusili prostě Set credentials…? (pak ještě to je nahoře v tom ToC)

Reply:
Je fakt, že upload evokuje něco jiného.